src/lib/healthCheck/subServiceHealth.js

```diff
@@ -0,0 +1,84 @@
+
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Lewis Daly <lewis@vesselstech.com>
+ --------------
+ ******/
+'use strict'
+
+const mongoose = require('mongoose')
+const { statusEnum, serviceName } = require('@mojaloop/central-services-shared').HealthCheck.HealthCheckEnums
+const Logger = require('@mojaloop/central-services-shared').Logger
+
+const Consumer = require('../kafka/consumer')
+
+/**
+ * @function getSubServiceHealthBroker
+ *
+ * @description Gets the health for the broker
+ * @returns Promise<SubServiceHealth> The SubService health object for the broker
+ */
+const getSubServiceHealthBroker = async () => {
+  const consumerTopics = Consumer.getListOfTopics()
+  let status = statusEnum.OK
+  try {
+    await Promise.all(consumerTopics.map(t => Consumer.isConsumerConnected(t)))
```

Comment:
I am not sure we should be using `Consumer.isConsumerConnected`. I believe this should be `Consumer.isConnected`, and similarly, this would be the same for the Producer (i.e. `Producer.isConnected`).

Having the name of the Class within the Method of a class is not a recommendation naming approach.

Reply:
Yeah that's a good point. I'll tidy up these names.